migrations/back-37/back-37.go

```diff
@@ -0,0 +1,207 @@
+	m.CLI().Authors = []cli.Author{
+		{
+			Name:  "J H BATE",
+			Email: "jamie@tidepool.org",
+		},
+	}
+
+	m.CLI().Action = func(ctx *cli.Context) error {
+		if !m.ParseContext(ctx) {
+			return nil
+		}
+		return m.execute()
+	}
+
+	return nil
+}
+
+func (m *Migration) execute() error {
+	m.Logger().Debug("Migrate jellyfish upload API data")
+	m.Logger().Debug("Creating data store")
+
+	mongoConfig := m.NewMongoConfig()
+	mongoConfig.Database = "data"
+	mongoConfig.Timeout = 60 * time.Minute
+	dataStore, err := storeStructuredMongo.NewStore(mongoConfig)
+	if err != nil {
+		return errors.Wrap(err, "unable to create data store")
+	}
+	defer dataStore.Terminate(context.Background())
+
+	m.Logger().Debug("Creating data repository")
+
+	m.dataRepository = dataStore.GetRepository("deviceData")
+	hashUpdatedCount, archivedCount, errorCount := m.migrateJellyfishDocuments()
+	m.Logger().Infof("Migrated %d jellyfish documents", hashUpdatedCount)
+	m.Logger().Infof("Archived %d duplicate jellyfish documents", archivedCount)
+	m.Logger().Infof("%d errors occurred", errorCount)
+
+	return nil
+}
+
+func (m *Migration) migrateJellyfishDocuments() (int, int, int) {
+	logger := m.Logger()
+
+	logger.Debug("Finding distinct users")
+
+	var hashUpdatedCount, archivedCount, errorCount int
+
+	userIDs, err := m.dataRepository.Distinct(context.Background(), "_userId", bson.M{})
+	if err != nil {
+		logger.WithError(err).Error("Unable to execute distinct query")
+	} else {
+		logger.Debugf("Finding jellyfish records for %d users", len(userIDs))
+
+		for _, userID := range userIDs {
+			logger.Debugf("Finding jellyfish records for user ID %s", userID)
+			selector := bson.M{
+				"_userId": userID,
+				"_active": true,
+				// uploads aren't de-duped.
+				"type":          bson.M{"$ne": "upload"},
+				"_deduplicator": bson.M{"$exists": false},
+			}
+
+			var jellyfishResult bson.M
+			jellyfishDocCursor, err := m.dataRepository.Find(context.Background(), selector)
+			if err != nil {
+				logger.WithError(err).Error("Unable to find jellyfish results")
+			}
+			for jellyfishDocCursor.Next(context.Background()) {
+				err = jellyfishDocCursor.Decode(&jellyfishResult)
+				if err != nil {
+					logger.WithError(err).Error("Could not decode mongo doc")
+					errorCount++
+					continue
+				}
+
+				dupQuery := bson.M{
+					"_userId":       userID,
+					"_active":       true,
+					"_id":           jellyfishResult["_id"],
+					"_deduplicator": bson.M{"$exists": true},
+				}
+				dupCursor, err := m.dataRepository.Find(context.Background(), dupQuery)
+				if err != nil {
+					logger.WithError(err).Errorf("Could not query for duplicate datum %s.", jellyfishResult["_id"])
+					errorCount++
+					continue
+				}
+				if dupCursor.Next(context.Background()) {
+					if !m.DryRun() {
+						if err := m.archiveDocument(jellyfishResult["_id"]); err != nil {
+							logger.WithError(err).Error("Unable to archive jellyfish document")
+							errorCount++
```

Comment:
If it errored, then should it count add to the `archivedCount`? Same as below with `hashUpdatedCount`.